backend/api/Controllers/MissionSchedulingController.cs

```diff
@@ -25,13 +25,80 @@ ISourceService sourceService
         ) : ControllerBase
     {
 
+        /// <summary>
+        ///     Rerun a mission run, running only the parts that did not previously complete
+        /// </summary>
+        /// <remarks>
+        ///     <para> This query runs the unfinished tasks of a previous mission run </para>
+        /// </remarks>
+        [HttpPost("rerun/{missionRunId}")]
+        [Authorize(Roles = Role.User)]
+        [ProducesResponseType(typeof(MissionRun), StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<MissionRun>> Rerun(
+            [FromRoute] string missionRunId,
+            [FromBody] ScheduleMissionQuery scheduledMissionQuery
+        )
+        {
+            var robot = await robotService.ReadById(scheduledMissionQuery.RobotId);
+            if (robot is null) return NotFound($"Could not find robot with id {scheduledMissionQuery.RobotId}");
+
+            var missionRun = await missionRunService.ReadById(missionRunId);
+            if (missionRun == null) return NotFound("Mission run not found");
+
+            var missionTasks = missionRun.Tasks.Where((t) => t.Status != Database.Models.TaskStatus.Successful && t.Status != Database.Models.TaskStatus.PartiallySuccessful).Select((t) => new MissionTask(t, Database.Models.TaskStatus.NotStarted)).ToList();
+
+            if (missionTasks == null || missionTasks.Count == 0) return NotFound("No unfinished mission tasks were found for the requested mission");
+
+            foreach (var task in missionTasks)
+            {
+                task.Id = Guid.NewGuid().ToString();
+                foreach (var inspection in task.Inspections)
+                {
+                    inspection.Id = Guid.NewGuid().ToString();
+                }
+            }
+
+            var newMissionRun = new MissionRun
+            {
+                Name = missionRun.Name,
+                Robot = robot,
+                MissionId = missionRun.MissionId,
+                Status = MissionStatus.Pending,
+                MissionRunPriority = MissionRunPriority.Normal,
+                Tasks = missionTasks,
+                DesiredStartTime = scheduledMissionQuery.DesiredStartTime ?? DateTime.UtcNow,
+                InstallationCode = missionRun.InstallationCode,
+                Area = missionRun.Area,
+                Map = new MapMetadata(missionRun.Map)
```

Comment:
```suggestion
                Map = missionRun.Map
```

Reply:
Since this is an owned property, we cannot copy the reference to it. Instead we need to make a copy of the map, which is done in the constructor.